### PR TITLE
pacific: osd/scrub: separate between PG state flags and internal scrubber operation

### PR DESCRIPTION
**Pull request by satoru-takeuchi**
backport tracker: https://tracker.ceph.com/issues/50900
backport of https://github.com/ceph/ceph/pull/41258
parent tracker: https://tracker.ceph.com/issues/50446

backport tracker: https://tracker.ceph.com/issues/51766
backport of https://github.com/ceph/ceph/pull/42410
parent tracker: https://tracker.ceph.com/issues/51581

### CODE REVIEW

**Comment by ronen-fr** [comment]
LGTM - apart from containing the original Test_auto_repair_bluestore_tag() that is known to have
a problem. I hope to issue a PR to fix that in 'master' today.

File: qa/standalone/scrub/osd-scrub-repair.sh [hunk 1]
```diff
@@ -318,6 +318,127 @@ function TEST_auto_repair_erasure_coded_overwrites() {
...
+    if [[ $st =~ (.*scrubbing.*) ]]
+    then
+        # a timeout
+        return 1
+    fi
+    return 0
+}
+
+
+function TEST_auto_repair_bluestore_tag() {
```
Comment (neha-ojha) [inline qa/standalone/scrub/osd-scrub-repair.sh:387]: Needs follow-on fix for https://tracker.ceph.com/issues/51581

**Comment by neha-ojha** [comment]
@satoru-takeuchi could you please include the follow-on fix https://tracker.ceph.com/issues/51766 in this PR?

**Comment by satoru-takeuchi** [comment]
@neha-ojha got it, I'll update this PR soon.

**Comment by neha-ojha** [comment]
> @neha-ojha got it, I'll update this PR soon.

thanks!

**Comment by satoru-takeuchi** [comment]
@ronen-fr @neha-ojha 

> could you please include the follow-on fix https://tracker.ceph.com/issues/51766 in this PR?

Done. Could you take a look at this PR again? I changed the PR description to mention 51766 as well.

**Review by neha-ojha — APPROVED** [review]

**Comment by yuriw** [comment]
jenkins test make check

**Pull request merged**